pkl/private/pkl.bzl

```diff
@@ -191,18 +191,19 @@ def _pkl_eval_impl(ctx):
 
     if ctx.attr.multiple_outputs:
         if len(ctx.attr.outs) > 0:
-            for output_name in ctx.attr.outs:
-                relative_path = "{}/{}".format(ctx.label.name, output_name)
-                outputs.append(ctx.actions.declare_file(relative_path))
-            output_location = outputs[0].path.removesuffix(ctx.attr.outs[0])
+            for output in ctx.outputs.outs:
+                outputs.append(output)
+            output_location = _find_common_prefix([o.path for o in outputs])
```

Comment:
This line finds the common path ancestor, so that we can pass it to `pkl eval` with `--output-path` (see `pkl/private/run_pkl_script.sh`). If there's a better way to do this, let me know.

Reply:
Let's add another test case that has at least one directory in common, e.g. `cars/fiat.yaml` and `cars/ford.json`.

With the suggested implementation, the common part is duplicated and we end up with `cars/cars/fiat.yaml` and `cars/cars/ford.json` output path will end up being duplicated.

The existing implementation (`outputs[0].path.removesuffix(ctx.attr.outs[0])`) is not pretty but ensures that the output location is always the bazel package dir.

Reply:
> Let's add another test case that has at least one directory in common, e.g. cars/fiat.yaml and cars/ford.json.

This was an excellent suggestion, that showed up issues with the previous implementation. I've now produced something that is much simpler and easier to understand for the multiple files with overlapping directories.